### ISSUE TITLE
LOC altitude should be float

### ISSUE THREAD

**Issue by hmoffatt** [StackExchange/dnscontrol#3125]
**Describe the bug**
The altitude field in the LOC format (https://docs.dnscontrol.org/language-reference/domain-modifiers/loc) is listed as uint32, but that's the encoded wire format. It's actually int32, and negative values work as expected.

However, the wire format allows the altitude to be specified down to centimetres, but the parser in `t_loc.go` supports int32 only.

**Comment by tlimoncelli** [comment]
CC @systemcrash

**Comment by systemcrash** [comment]
Sort of. int32 handling would truncate max altitude to something lilke 21374836,47m, so it must be uint32 to accommodate the maximum altitude of 42849672.95m. But you're correct in noticing that the decimal portion of the altitude wasn't correctly handled. The parser shall be updated.

It seems I overlooked handling cm values, i.e. the decimal portion (expressed as e.g. `0.45m`).

I have proposed a fix in #3130 - you're welcome to test it locally and confirm that it resolves you're issues (I've tested it on my provider and everything seems to pass).
